### PR TITLE
Fix #1795: Validate user ID for operation approve and reject before updating operation

### LINKED ISSUE #1795: Validate user ID for operation approve and reject before updating operation
Move user ID validation before updating the operation failed attempts to avoid potential manipulation with operations of other users, which can lead to denial of service.

### PR DESCRIPTION
**Pull request by romanstrobl**

### CODE REVIEW

**Review by banterCZ — DISMISSED** [review]
Looks good.

File: powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/OperationServiceBehavior.java [hunk 1]
```diff
@@ -375,13 +375,17 @@ public OperationUserActionResponse attemptApproveOperation(OperationApproveReque
                 throw localizationProvider.buildExceptionForCode(ServiceError.OPERATION_APPROVE_FAILURE);
             }
 
+            if (!expectedUserId.equals(userId)) {
+                logger.warn("Operation ID: {} cannot be approved, because user ID from the request '{}' does not match user ID from the operation '{}'.", operationId, userId, expectedUserId);
```
Comment (zcgandcomp) [inline powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/OperationServiceBehavior.java]: can we switch to pattern operationId (this is more easily searchable in logs) Thank you.

**Review by zcgandcomp — DISMISSED** [review]
Over all good. 
Correct returned exception and we corrected previously incorrect exception from reject.

We should start using pattern `operationId` to have easier parsable logs.

File: powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/OperationServiceBehavior.java (hunk 1 above)
Comment (romanstrobl) [reply powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/OperationServiceBehavior.java]: Ok, I will update it.

**Review by banterCZ — APPROVED** [review]

**Pull request merged**